code-challenges.test.js

```diff
@@ -13,40 +13,176 @@
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
 // a) Create a test with expect statements using the variables provided.
-
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("changer", () => {
+    it("takes in a string and returns a string with a coded message", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        const output1 = "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        const output2 = "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        const output3 = "3cc3ntr1c"
+      expect(changer(secretCodeWord1)).toEqual(output1)
+      expect(changer(secretCodeWord2)).toEqual(output2)
+      expect(changer(secretCodeWord3)).toEqual(output3)
+    })
+  })
+
+// ReferenceError: changer is not defined
 
 // b) Create the function that makes the test pass.
 
+//create a function called changer
+//will take the parameter of string
+  //function will go through each letter of the string
+  //the function will change each vowel to a specific number
+  //will return the new string
+//test function
```

Comment:
There is a lot more going on in your code than what is provided in your pseudocode.  I would really like to see more explanation on what your process is.  For instance, iteration and some of the different built in methods being used.